File: frontend/planta-smart-4.0/src/infrastructure/dtos/response/stream/FilaStreamDTO.ts
import { PedidoGetResponseDTO } from '@dtos/response/PedidoGetResponseDTO';

// Espelha o FilaStreamDTO do backend (snapshot da fila de produção).
// Recebido tanto via REST (GET /api/smart/fila) quanto via SSE (evento "fila").
export interface FilaStreamDTO {
  emExecucao: PedidoGetResponseDTO | null;
  tempoExecucaoSegundos: number;
  pendentes: PedidoGetResponseDTO[];
}

File: backend/planta-smart-4.0/src/main/java/com/smart/appsa/dto/response/FilaStreamDTO.java
package com.smart.appsa.dto.response;

import java.util.List;

// Snapshot da fila de produção exposto via REST e SSE.
// {@code emExecucao} é o pedido sendo produzido (ou {@code null}); {@code pendentes}
// são os enfileirados em ordem FIFO; {@code tempoExecucaoSegundos} é o tempo decorrido
// do pedido em execução (0 quando não há pedido em curso).
public record FilaStreamDTO(
        PedidoResponseDTO emExecucao,
        long tempoExecucaoSegundos,
        List<PedidoResponseDTO> pendentes) {
}
